subprojects/core/src/main/java/org/gradle/api/internal/initialization/StandaloneDomainObjectContext.java

```diff
@@ -26,18 +28,52 @@
 import java.util.function.Consumer;
 import java.util.function.Function;
 
-public class RootScriptDomainObjectContext implements DomainObjectContext, ModelContainer<Object> {
+public abstract class StandaloneDomainObjectContext implements DomainObjectContext, ModelContainer<Object> {
```

Comment:
🤔 As discussed, perhaps `NonProjctDomainObjectContext would be a better name.

Reply:
Given that we will eventually want to have `Settings` also be a domain object context, it feels weird to single-out `NonProject`

Reply:
That's true.  Maybe just something to add to javadoc on this type.